vunit/sim_if/ghdl.py

```diff
@@ -364,3 +392,60 @@ def simulate(  # pylint: disable=too-many-locals
             subprocess.call(cmd)
 
         return status
+
+    def _compile_source_file(self, source_file, printer):
+        """
+        Runs parent command for compilation, and moves any .gcno files to the compilation output
+        """
+        compilation_ok = super()._compile_source_file(source_file, printer)
+
+        # GCOV gcno files are output to where the command is run,
+        # move it back to the compilation folder
+        source_path = Path(source_file.name)
+        gcno_file = Path(source_path.stem + ".gcno")
+        if Path(gcno_file).exists():
+            new_path = Path(source_file.library.directory) / gcno_file
+            gcno_file.rename(new_path)
+
+        return compilation_ok
+
+    def merge_coverage(self, file_name, args=None):
+        """
+        Merge coverage from all test cases
+        """
+        output_dir = file_name
+
+        # Loop over each .gcda output folder and merge them two at a time
+        first_input = True
+        for coverage_dir in self._coverage_test_dirs:
+            if file_exists(coverage_dir):
+                if first_input:
+                    # There is no output to merge with, so merge with itself to produce first output
+                    first_dir = coverage_dir
+                else:
+                    first_dir = output_dir
+                merge_command = [
+                    "gcov-tool",
+                    "merge",
+                    "-o",
+                    output_dir,
+                    first_dir,
+                    coverage_dir,
+                ]
+                subprocess.call(merge_command)
+                first_input = False
+            else:
+                LOGGER.warning("Missing coverage directory: %s", coverage_dir)
+
+        # Find actual output path of the .gcda files (they are deep in hierarchy)
+        dir_path = Path(output_dir)
+        gcda_files = dir_path.glob("**/*.gcda")
+        gcda_dirs = [x.parent for x in gcda_files]
```

Comment:
Maybe replace 442-444 with `gcda_dirs = set([x.parent for x in dir_path.glob("**/*.gcda")])`?

Reply:
Done, thanks. Actually, replaced with set comprehension after recommendation from linter: `gcda_dirs = {[x.parent for x in dir_path.glob("**/*.gcda")}`